tests/wpt/mozilla/tests/mozilla/xmlhttprequest_url.html

```diff
@@ -0,0 +1,27 @@
+<!doctype html>
+<meta charset="utf-8">
+<title>XMLHttpRequest Origin</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+function resolve(path) {
+  let a = document.createElement("a");
+  a.href = path;
+  return a.href;
+}
+
+onload = function () {
```

Comment:
No need to wait for onload here. We can write this test like this:
```js
async_test(function(t) {
  let href = window.location.href;
  let request = new XMLHttpRequest();
  let url = resolve("./test.txt");
  request.open('GET', url, true);

  request.onload = t.step_func_done(function() {
    let entries = window.performance.getEntriesByType("resource");
    assert_equals(entries.length, 1);
    assert_equals(entries[0].name, href);
  });
}, "Performance entries should contain the URL where the XMLHttpRequest originated");
```
I'm a bit surprised that the test passes right now, since I would expect the test harness `<script>` elements to be included in the resulting performance entries :<

Reply:
Hmm, I'm not sure why it doesn't show the performance entries for script elements. Calling `getEntries()` only shows the new xmlhttprequest as well. I am wondering whether it would be better to clear out the resource entries beforehand just in case: `window.performance.clearResourceTimings();`.
Or, maybe get the names of the entries first and then test for the new entry, Something like:
```
   let entries = window.performance.getEntriesByType("resource");
   let nameEntries = [];
   entries.forEach(entry => {
       nameEntries.push(entry.name);
   });
   assert_true(nameEntries.includes(href));
```
As far as the entries for the script elements, again I am not sure why but I can try to find out what may be the issue there.

Reply:
I would rather have the test yell at us if we change the implementation in a way that makes it no longer report expected results. Let's merge it as-is.